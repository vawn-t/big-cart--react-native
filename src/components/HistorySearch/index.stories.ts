import type { Meta, StoryObj } from '@storybook/react-native';

import HistorySearch from '.';

const meta: Meta<typeof HistorySearch> = {
  title: 'Components/HistorySearch',
  component: HistorySearch,
};

export default meta;

type Story = StoryObj<React.ComponentProps<typeof HistorySearch>>;

export const Base: Story = {
  args: {},
};
