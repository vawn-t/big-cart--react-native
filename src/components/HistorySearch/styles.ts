import { StyleSheet } from 'react-native';
import { colors } from '@themes/index';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    rowGap: 15,
  },
  titleWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  clear: {
    color: colors.link,
  },
  itemWrapper: {
    gap: 10,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  item: {
    backgroundColor: colors.background.primary,
    paddingVertical: 6,
    paddingHorizontal: 8,
  },
});

export default styles;
