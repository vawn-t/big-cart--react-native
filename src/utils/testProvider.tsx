import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { RenderOptions, render } from '@testing-library/react-native';

const Stack = createNativeStackNavigator();
const queryClient = new QueryClient();

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const component = () => <>{children}</>;

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="test">{component}</Stack.Screen>
        </Stack.Navigator>
      </NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  // eslint-disable-next-line react/display-name
  return ({ children }: { children: React.ReactNode }) => {
    const component = () => <>{children}</>;
    return (
      <QueryClientProvider client={queryClient}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="test">{component}</Stack.Screen>
          </Stack.Navigator>
        </NavigationContainer>
      </QueryClientProvider>
    );
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render, createWrapper };
