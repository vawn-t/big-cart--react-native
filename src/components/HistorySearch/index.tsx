import { useCallback } from 'react';
import { TouchableOpacity, View } from 'react-native';

// Components
import { Typography } from '@components/common';

// Types
import { FontWeight, SizeType, TypoVariant } from '@interfaces/index';

// Stores
import useStore from '@stores/useStore';

// Styles
import styles from './styles';

const HistorySearch = () => {
  // Stores
  const searchHistory = useStore.use.searchHistory();
  const resetSearchHistory = useStore.use.resetSearchHistory();
  const setCurrentSearchItem = useStore.use.setCurrentSearchItem();

  const handleSelectSearchHistory = useCallback((item: string) => {
    setCurrentSearchItem(item);
  }, []);

  const handleClearSearchHistory = useCallback(() => {
    resetSearchHistory();
  }, []);

  return (
    <View style={styles.container}>
      <View style={styles.titleWrapper}>
        <Typography
          text="Search History"
          variant={TypoVariant.Paragraph2}
          size={SizeType.Large}
          fontWeight={FontWeight.SemiBold}
        />

        <Typography
          text="clear"
          size={SizeType.Small}
          style={styles.clear}
          onPress={handleClearSearchHistory}
        />
      </View>
      <View style={styles.itemWrapper}>
        {searchHistory.map((item, index) => (
          <TouchableOpacity
            hitSlop={{ top: 10, left: 10, right: 10, bottom: 10 }}
            onPress={() => handleSelectSearchHistory(item)}
            key={`${item}-${index}`}
          >
            <Typography text={item} style={styles.item} size={SizeType.Xs} />
          </TouchableOpacity>
        ))}
      </View>
    </View>
  );
};

export default HistorySearch;
