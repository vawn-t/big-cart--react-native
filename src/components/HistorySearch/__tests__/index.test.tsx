import { render } from '@utils/testProvider';
import HistorySearch from '..';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

describe('HistorySearch component', () => {
  const setup = () => {
    return render(<HistorySearch />);
  };

  it('renders correctly', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('triggers the "handleClearSeachHistory" function when "clear" is pressed', () => {
    // TODO: should do
  });
});
