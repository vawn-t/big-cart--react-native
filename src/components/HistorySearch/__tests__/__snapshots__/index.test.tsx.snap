// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistorySearch component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "rowGap": 15,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                    {
                      "fontFamily": "Poppins-SemiBold",
                    },
                    {
                      "fontSize": 20,
                    },
                    undefined,
                  ]
                }
              >
                Search History
              </Text>
              <Text
                onPress={[Function]}
                style={
                  [
                    {
                      "color": "#868889",
                    },
                    {
                      "fontFamily": "Poppins-Regular",
                    },
                    {
                      "fontSize": 12,
                    },
                    {
                      "color": "#407EC7",
                    },
                  ]
                }
              >
                clear
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "gap": 10,
                }
              }
            />
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
